### PR TITLE
Add tox testing and coverage

### PR DESCRIPTION
**Pull request by quantumjot**
+ [x] add tox for testing on win, mac os and ubuntu for py 3.7-3.9
+ [x] add codecov report
+ [ ] add badges to README

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+PLATFORM =
+    ubuntu-latest: linux
+    macos-latest: macos
+    windows-latest: windows
+
+[testenv]
+platform =
+    macos: darwin
+    linux: linux
+    windows: win32
```
Comment (paddyroddy) [inline tox.ini:21]: Why test on Windows 32bit?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -4,22 +4,35 @@ on:
...
+          pip install setuptools tox tox-gh-actions
+
+      # this runs the platform-specific tests declared in tox.ini
+      - name: Test with tox
+        run: tox
+        env:
+          PLATFORM: ${{ matrix.platform }}
+
+      - name: Coverage
+        uses: codecov/codecov-action@v2
```
Comment (paddyroddy) [inline .github/workflows/build.yml:38]: Very recently released
```suggestion
        uses: codecov/codecov-action@v3
```

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -4,22 +4,35 @@ on:
...
-        python -m pip install --upgrade pip
-    - name: Test with pytest
-      run: |
-        pip install --upgrade --upgrade-strategy eager -r requirements.txt -e .
-        pip install pytest
-        pytest
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
```
Comment (paddyroddy) [inline .github/workflows/build.yml:18]:
```suggestion
        uses: actions/setup-python@v3
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -4,22 +4,35 @@ on:
...
-    - name:
-        Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-    - name: Test with pytest
-      run: |
-        pip install --upgrade --upgrade-strategy eager -r requirements.txt -e .
-        pip install pytest
-        pytest
+      - uses: actions/checkout@v2
```
Comment (paddyroddy) [inline .github/workflows/build.yml:15]:
```suggestion
      - uses: actions/checkout@v3
```

**Review by paddyroddy — APPROVED** [review]
Looks good, just some suggestions

File: tox.ini (hunk 1 above)
Comment (quantumjot) [reply tox.ini:21]: good question! will change to `windows` -- this will prob be true for the other plugins too.

Comment (quantumjot) [reply tox.ini]: done!

**Pull request merged**